source/controllers/sudokuWorld.d

```diff
@@ -0,0 +1,161 @@
+	{
+		trace("Initializing SudokuWorld callbacks...");
+		app.setOnBtnCreateClicked(&onBtnCreateClicked);
+		app.setOnBtnExitClicked(&onBtnExitClicked);
+		app.setOnButtonPress(&onButtonPress);
+	}
+
+
+	/** This is called when `btnCreate` is clicked in the main menu
+	 *
+	 * Makes the create menu content `Stack` visible
+	 */
+	private void onBtnCreateClicked(Button)
+	{
+		app.showCreateMenu();
+	}
+
+
+	/** Exit the program
+	 */
+	private void onBtnExitClicked(Button)
+	{
+		app.quit();
+	}
+
+
+	// TODO: ui:sudokuWorld: implement Play Menu
+	private void onBtnPlayClicked(Button)
+	{
+		// do something
+	}
+
+
+	// TODO: ui:sudokuWorld: implement Settings Menu
+	private void onBtnSettingsClicked(Button)
+	{
+		// do something
+	}
+
+
+	private bool onButtonPress(GdkEventButton* ev, Widget widget)
+	{
+		if (ev.button != 1)
+			return false;
+
+
+		// clear focus in the active grid
+		GridUI gridUI;
+		if (GameState.Play in sudokuControllers)
+		{
+			gridUI = sudokuControllers[GameState.Play].gridUI;
+		}
+		else if (GameState.Solution in sudokuControllers)
+		{
+			gridUI = sudokuControllers[GameState.Solution].gridUI;
+			_createMenuController.setCellInfoText(0,0,0);
+		}
+		else if (GameState.Create in sudokuControllers)
+		{
+			gridUI = sudokuControllers[GameState.Create].gridUI;
+			_createMenuController.setCellInfoText(0,0,0);
+		}
+		gridUI.cleanFocus();
+		return false;
+	}
+
+
+
+// Getters/Setters
+
+	public void addSudokuController(SudokuController sudokuController)
+	{
+		sudokuControllers[sudokuController.gameState] = sudokuController;
+	}
+
+
+	public auto addSudokuController(GameState state, SudokuType type)
+	{
+		Grid grid = new Grid(Sudoku.dimension(type).expand);
+		grid.initialize(new int[][](grid.height, grid.width));
+		GridUI gridUI = new GridUI(type);
+		SudokuController sd = new SudokuController(grid, gridUI, state, this);
+		return sudokuControllers[state] = sd;
+	}
+
+
+	public auto addSudokuController(GameState state, SudokuController sudokuController)
+	{
+		SudokuType type = sudokuController.gridUI.type;
+		Grid grid = new Grid(Sudoku.dimension(type).expand);
+		int[][] digits = sudokuController.grid.toDigit();
+		grid.initialize(digits.dup);
+		GridUI gridUI = new GridUI(type);
+		SudokuController sd = new SudokuController(grid, gridUI, state, this);
+		return sudokuControllers[state] = sd;
+	}
+
+
+	public void deleteSudokuController(GameState state)
+	{
+		SudokuController* p = (state in sudokuControllers);
+		if (p !is null)
+		{
+			SudokuController sc = *p;
+			sc.gridUI.destroy();
+			sudokuControllers.remove(state);
+		}
+	}
+
+
+	public auto getSudokuController(GameState state)
+	{
+		return sudokuControllers[state];
+	}
+
+
+	public auto createMenuController() @property
+	{
+		return _createMenuController;
+	}
+
+
+	public auto app() @property
+	{
+		return _app;
+	}
+
+
+	private CreateMenuController _createMenuController;
+	private SudokuController[string] sudokuControllers;
```

Comment:
Yes you're right. I made it that way in the beginning because I was thinking of using custom names. Then I implemented `GameState`, which is an `enum : string` and forgot to change it as it is the same value type in the end.